lib/ansible/module_utils/basic.py

```diff
@@ -2126,12 +2118,12 @@ def preexec():
             selector.close()
 
             rc = cmd.returncode
-        except (OSError, IOError) as e:
-            self.log("Error Executing CMD:%s Exception:%s" % (self._clean_args(args), to_native(e)))
+        except OSError as ex:
+            self.log(f"Error Executing CMD:{self._clean_args(args)} Exception: {ex}")
             if handle_exceptions:
-                self.fail_json(rc=e.errno, stdout=b'', stderr=b'', msg=to_native(e), cmd=self._clean_args(args))
+                self.fail_json(rc=ex.errno, stdout=b'', stderr=b'', msg=to_native(ex), cmd=self._clean_args(args), exception=ex)
```

Comment:
These won't stay bytes anyways, so just use strings. Also, we might as well include a slightly more helpful msg instead of just repeating the exception text:

```suggestion
                self.fail_json(rc=ex.errno, stdout='', stderr='', msg="Error executing command.", cmd=self._clean_args(args), exception=ex)
```

We could even include the command if we're confident in the `_clean_args` logic.

I'd fix up the one below too -- I mention it here since GitHub won't let me comment on it there.